src/com/intellij/idea/plugin/hybris/toolwindow/RemoteHacConnectionDialog.kt

```diff
@@ -144,6 +158,60 @@ class RemoteHacConnectionDialog(
                     .onChanged { urlPreviewLabel.text = generateUrl() }
                     .component
             }.layout(RowLayout.PARENT_GRID)
+            if (System.getProperty("os.name").lowercase(Locale.getDefault()).contains("win")) {
+                val distributions = WslDistributionManager.getInstance().installedDistributions
+                row {
+                    isWslCheckBox = checkBox("WSL")
+                        .selected(false)
+                        .visible(distributions.isNotEmpty())
+                        .onChanged {
+                            val selected = isWslCheckBox.isSelected
+                            val multipleDistros = distributions.isNotEmpty()
+                            wslDistributionComboBox.isVisible = selected && multipleDistros
+                            wslDistributionText.visible(selected)
+                            wslProxyCheckBox.isVisible = selected
+                            wslProxyWarningComment.isVisible = selected
+                            urlPreviewLabel.text = generateUrl()
+                        }
+                        .component
+                }.layout(RowLayout.PARENT_GRID)
+                val installedDistros = distributions.map { it.msId }
+                if (installedDistros.isNotEmpty()) {
+                    row {
+                        wslDistributionText = label("WSL distribution:").visible(false)
+                        wslDistributionComboBox = comboBox(DefaultComboBoxModel(installedDistros.toTypedArray()))
+                            .align(AlignX.FILL)
+                            .visible(false)
+                            .onChanged {
+                                updateWslIp(distributions)
+                            }
+                            .component
+                    }.layout(RowLayout.PARENT_GRID)
+                } else {
+                    row {
+                        comment("No WSL distributions are installed.")
+                            .visible(false)
+                            .component
+                    }.layout(RowLayout.PARENT_GRID)
+                }
+                row {
+                    wslProxyCheckBox = checkBox("Enable wsl.proxy.connect.localhost")
+                        .comment("This will use the wsl.proxy.connect.localhost registry setting if available.")
+                        .visible(false)
+                        .selected(Registry.`is`("wsl.proxy.connect.localhost"))
+                        .onChanged {
+                            Registry.run { get("wsl.proxy.connect.localhost").setValue(!`is`("wsl.proxy.connect.localhost")) }
```

Comment:
extract `wsl.proxy.connect.localhost` to a constant

Reply:
Completed